src/with-filter.ts

```diff
@@ -1,10 +1,8 @@
-import { $$asyncIterator } from 'iterall';
-
 export type FilterFn<TSource = any, TArgs = any, TContext = any> = (rootValue?: TSource, args?: TArgs, context?: TContext, info?: any) => boolean | Promise<boolean>;
-export type ResolverFn<TSource = any, TArgs = any, TContext = any> = (rootValue?: TSource, args?: TArgs, context?: TContext, info?: any) => AsyncIterator<any>;
+export type ResolverFn<TSource = any, TArgs = any, TContext = any> = (rootValue?: TSource, args?: TArgs, context?: TContext, info?: any) => AsyncIterator<any> | Promise<AsyncIterator<any>>;
```

Comment:
I think to have proper typing there needs to be a distinction between the `asyncIteratorFn` received as input which needs to return an `AsyncIterator` and the `withFilter`method that actually returns a `AsyncIterableIterator`

Should split `ResolverFn` in 2 types to make the distinction.

Reply:
Added in #276